pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -183,25 +220,54 @@ func (s *LogStore) storeEntriesAndCommitBatch(
 	// (Replica), so this comment might need to move.
 	stats.PebbleBegin = timeutil.Now()
 	stats.PebbleBytes = int64(batch.Len())
-	sync := rd.MustSync && !disableSyncRaftLog.Get(&s.Settings.SV)
-	if err := batch.Commit(sync); err != nil {
-		const expl = "while committing batch"
-		return RaftState{}, errors.Wrap(err, expl)
+	mustSync := len(m.Responses) > 0
```

Comment:
Maybe this dance between things we must syncs and refusing to sync due to the `disableSyncRaftLog` setting could a bit clearer if we renamed `mustSync` -> `wantsSync` and `sync` -> `willSync`. Just a thought, take it or leave it.